driver_go18_test.go

```diff
@@ -189,27 +160,40 @@ func testCancel(dbt *DBTest, ctx context.Context, cancel context.CancelFunc, que
 	}
 
 	errChan := make(chan error)
-	go checkCancellation(dbt.db, dbname, query, cancel, errChan, getIdTimeout, killTimeout, pollTimeout)
-
-	// This query will be canceled.
-	startTime := time.Now()
-	err = queryFunc()
-	if err != nil && err != context.Canceled {
-		errLog.Print(err)
-	}
+	go func() {
+		// This query will be canceled.
+		startTime := time.Now()		
+		err = queryFunc()
+		if err != nil && err != context.Canceled {
+			errLog.Print(err)
+		}
+		if err != context.Canceled && ctx.Err() != context.Canceled {
+			errChan <- fmt.Errorf("expected context.Canceled, got %v", err)
+		}
+		if d := time.Since(startTime); d > 500*time.Millisecond {
+			errChan <- fmt.Errorf("too long execution time: %s", d)
+		}
+		errChan <- nil
+	}()
 
-	if err != context.Canceled && ctx.Err() != context.Canceled {
-		dbt.Errorf("expected context.Canceled, got %v", err)
+	// it is safe to not use timeouts here since they are inside the killQuery function
+	err = killQuery(dbt.db, dbname, query, cancel)
+	if err != nil {
+		dbt.Error(err)
+		return
 	}
 
-	if d := time.Since(startTime); d > 500*time.Millisecond {
-		dbt.Errorf("too long execution time: %s", d)
+	select {
+	case <-time.After(15 * time.Second):
```

Comment:
seems like this is unreachable, because your kill query should fail, before.

Reply:
Yes, you're right. Fixed.